Sources/eppo/EppoClient.swift

```diff
@@ -833,6 +854,16 @@ public class EppoClient {
         poller?.stop()
         poller = nil
     }
+    
+    /// Registers a callback for when a new configuration is applied to the EppoClient instance.
+    public func onConfigurationChange(_ callback: @escaping ConfigurationChangeCallback) {
```

Comment:
would a dev ever need to unregister a callback?

Reply:
I can't think of any obvious reasons.

Given that it's simpler to add than remove interfaces in SDKs, I'd hold off on adding it until requested. This matches up with the android functionality https://github.com/Eppo-exp/android-sdk/blob/72b62da65c121e4cf982ede90b47ba0d6db8759f/eppo/src/main/java/cloud/eppo/android/EppoClient.java#L256-L262